subsys/suit/orchestrator/src/suit_orchestrator_sdfw.c

```diff
@@ -86,6 +88,44 @@ static void leave_emergency_recovery(void)
 	}
 }
 
+static bool is_address_range_allowed_for_update_region(const uint8_t *address, size_t size)
+{
+	/** The update region (update candidate or DFU cache) address is considered valid in three cases:
+	 * 1) address in external flash
+	 * 2) address accessible by application core
+	 * 3) address accessible by radio core
+	 *  
+	 * Other cases can lead to security issues, leading to potential leaks
+	 * of protected data.
+	 */
+	
+	if (suit_memory_global_address_range_is_in_external_memory((uintptr_t) address, size)) {
+		return true;
+	}
```

Comment:
The `update_region[0]` must be in MRAM. I would suggest  that this check is done only for the `update_regions[i > 0]`